pkg/operator/options/options.go

```diff
@@ -59,23 +85,101 @@ func New() *Options {
 	f.BoolVar(&opts.EnableLeaderElection, "leader-elect", env.WithDefaultBool("LEADER_ELECT", true), "Start leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.")
 	f.Int64Var(&opts.MemoryLimit, "memory-limit", env.WithDefaultInt64("MEMORY_LIMIT", -1), "Memory limit on the container running the controller. The GC soft memory limit is set to 90% of this value.")
 
+	// Vars that must be merged with settings
+	f.DurationVar(&opts.BatchMaxDuration, "batch-max-duration", env.WithDefaultDuration("BATCH_MAX_DURATION", 10*time.Second), "TODO")
+	f.DurationVar(&opts.BatchIdleDuration, "batch-idle-duration", env.WithDefaultDuration("BATCH_IDLE_DURATION", time.Second), "TODO")
+	f.StringVar(&opts.featureGates, "feature-gates", env.WithDefaultString("FEATURE_GATE", ""), "TODO")
+
 	if opts.MemoryLimit > 0 {
 		newLimit := int64(float64(opts.MemoryLimit) * 0.9)
 		debug.SetMemoryLimit(newLimit)
 	}
+
 	return opts
 }
 
-// MustParse reads the user passed flags, environment variables, and default values.
-// Options are valided and panics if an error is returned
 func (o *Options) MustParse() *Options {
-	err := o.Parse(os.Args[1:])
-
-	if errors.Is(err, flag.ErrHelp) {
+	if err := o.Parse(os.Args[1:]); errors.Is(err, flag.ErrHelp) {
 		os.Exit(0)
-	}
-	if err != nil {
+	} else if err != nil {
 		panic(err)
 	}
+
+	// Check if BatchMaxDuration and BatchIdleDuration have been set. If they have not been set, they will be overwritten by
+	// values in the configmap, if present.
+	o.Visit(func(f *flag.Flag) {
+		switch f.Name {
+		case "batch-max-duration":
+			o.BatchMaxDurationSet = true
+		case "batch-idle-duration":
+			o.BatchIdleDurationSet = true
+		}
+	})
+	if _, ok := os.LookupEnv("BATCH_MAX_DURATION"); ok {
+		o.BatchMaxDurationSet = true
+	}
+	if _, ok := os.LookupEnv("BATCH_IDLE_DURATION"); ok {
+		o.BatchIdleDurationSet = true
+	}
+
+	if o.featureGates != "" {
+		gates := parseFeatureGates(o.featureGates)
+		val, ok := gates["Drift"]
+		o.DriftEnabled = lo.Ternary(ok, val, false)
+		o.DriftEnabledSet = ok
+	}
+
 	return o
 }
+
+// MergeSettings applies settings specified in the v1alpha5 configmap to options. If the value was already specified by
+// a CLI argument or environment variable, that value will be used.
+func MergeSettings(ctx context.Context, s *settings.Settings) {
+	o := FromContext(ctx)
+	if !o.BatchMaxDurationSet {
+		o.BatchMaxDuration = s.BatchMaxDuration
+		o.BatchMaxDurationSet = true
+	}
+	if !o.BatchIdleDurationSet {
```

Comment:
Same comment here. We should check each setting is overridable individually, or consider specifying the defaults at the settings struct (or both places potentially) and only override the options if the option wasn't explicitly set by the user.

Reply:
Restoring the defaults to the settings struct makes sense to me, along with keeping the individual field check. I'll go ahead and do that.